### ISSUE TITLE
Add support for Sentry

### ISSUE THREAD

**Issue by adinauer** [spring-io/start.spring.io#1114]
Hello 👋, I'm Alex and I'm working on the [Sentry Java SDK](https://github.com/getsentry/sentry-java).

[Sentry](https://sentry.io/welcome) helps developers identify and fix their bugs as well as provide performance monitoring and distributed tracing.

Spring Boot is very popular with our users. We'd like to provide them an even better experience by being able to quickly add Sentry to their new project created via [start.spring.io](https://start.spring.io/).

Please allow us to add Sentry. We're happy to open a PR for adding Sentry here.

# Requirements
To ensure the ongoing health of the Spring Boot community, a starter that is included on
start.spring.io must meet the following requirements.

## Code of Conduct
This project adheres to the Contributor Covenant code of conduct. By participating, you
are expected to adhere to this code. If you have not already done so, we also encourage
you to adopt a code of conduct in your own project.

- https://github.com/getsentry/.github/blob/2312f1c9cb22b86afa2cc642c93b0b2457700697/CODE_OF_CONDUCT.md

## Well-established project with a vibrant community
To avoid overwhelming users with too much choice, we aim for each starter to be of
interest to a broad range of start.spring.io's users. One way to ensure this is for each
starter to represent a well-established project with a vibrant community.

Please provide some links that provide some evidence for this. We're looking for
things like GitHub stars, multiple contributors, ongoing maintenance, Stack Overflow
questions, etc.

- Java SDK GitHub repo: https://github.com/getsentry/sentry-java
- Sentry GitHub repo: https://github.com/getsentry/sentry

## Ongoing maintenance
To ensure that the Spring Boot community has a good experience with your starter,
it should be kept up-to-date with the latest Spring Boot releases and maintenance
releases should be shipped as needed. When a new release is made, a pull request
should be made to this project to ensure that the Spring Boot community are using the
latest version

- Releases: https://github.com/getsentry/sentry-java/releases
- Commits: https://github.com/getsentry/sentry-java/commits/main

## Licence
Please provide details of your project's licence. If you starter pulls in any
dependencies with a different license to the  starter's own licence, please provide
details of those dependencies' licences as well.

- The Java SDK is licensed under MIT: https://github.com/getsentry/sentry-java/blob/main/LICENSE
- https://github.com/getsentry/sentry/blob/master/LICENSE

## Issue Tracker URL
To allow users of your starter to contribute and provide feedback, please provide
the URL of its issue tracker.

- Java SDK GitHub issues: https://github.com/getsentry/sentry-java/issues
- Sentry GitHub issues: https://github.com/getsentry/sentry/issues

## Continuous integration
To help to ensure the ongoing compatibility of your starter with Spring Boot, please
provide a link to the CI jobs that verify its compatibility. Compatibility should
be achieved using Spring Boot's dependency management without overriding.

- https://github.com/getsentry/sentry-java/actions/workflows/build.yml

## Maven Central
Starters that are included on start.spring.io must be available on Maven Central.

[x] Starter and all of its dependencies are available on Maven Central.

## Configuration metadata
If you starter provides `@ConfigurationProperties` it should provide metadata for those
properties to enable auto-completion in the IDE. Metadata can be provided by adding
field level documentation on each of the properties and adding the
`spring-boot-configuration-processor` to your build.

[x] Starter provides metadata, including documentation, for all its configuration
properties.

## Version mappings
Spring Initializr allows different versions of a starter to be used with different
versions of Spring Boot. For example, if a project is being generated with Spring
Boot version W you may want to use Starter version X and if a project is being
generated with Spring Boot version Y you may want to use Starter version Z.

Please provide the details of any mappings between particular version of Spring
Boot and particular versions of your starter.

- Spring boot 3.x requires `artifactId` `sentry-spring-boot-starter-jakarta`
- Spring boot 2.1+ requires `artifactId` `sentry-spring-boot-starter`
- Both are released with the same version, so we support them in parallel

## Links to additional resources
Each entry can be associated with one or several links. Please provide at least a link to
a sample and the reference documentation:

* Reference documentation:
	* https://docs.sentry.io/platforms/java/guides/spring-boot/
* Sample:
	* For Spring Boot 3.x: https://github.com/getsentry/sentry-java/tree/main/sentry-samples/sentry-samples-spring-boot-jakarta
	* For Spring Boot 2.1+: https://github.com/getsentry/sentry-java/tree/main/sentry-samples/sentry-samples-spring-boot

**Comment by adinauer** [comment]
Hey, please let me know if there's anything I can do to help you process this proposal.

**Comment by HazAT** [comment]
Hey, I just wanted to ask if there is anything blocking this or are we free to submit and get a PR merged for Sentry 😄 ?
Thanks

**Comment by marandaneto** [comment]
> Thank you for the interest. We need time to review the proposal.

Do you need anything else from us to get this going? Thanks.

**Comment by snicoll** [comment]
@adinauer, @HazAT, and @marandaneto thank you for your patience.

When you get a chance, I've created https://github.com/getsentry/sentry-java/issues/2866 as the starter does not currently comply with our requirement. Nothing urgent, I've configured start.spring.io to add the core starter if Sentry is the only entry selected.

Congrats on a very well maintained getting started guide, it's great!

**Comment by adinauer** [comment]
Thank you very much @snicoll 🚀 I replied on the linked issue.
